src/Donations/Micros/Authentication.tsx

```diff
@@ -43,16 +44,33 @@ function Authentication({}: Props): ReactElement {
   const [openVerifyEmailModal, setopenVerifyEmailModal] = React.useState(false);
 
   const loadUserProfile = async () => {
-    if ((user && user.email_verified) || queryToken) {
+    let token;
+    if (queryToken) {
+      const jwtToken = parseJwt(queryToken);
+      const isTokenValid =
+        new Date(jwtToken.exp * 1000).getTime() > new Date().getTime();
+
+      if (isTokenValid) {
+        token = queryToken;
+      }
```

Comment:
This can lead to the case that `token` is not set, but `user && user.email_verified` are defined and the token is not retrieved by `getAccessTokenSilently()`. The `if` condition below still is valid and the profile request get executed with an undefined token! Please fix this logic!